src/server/pkg/testutil/kubeclient.go

```diff
@@ -146,3 +148,58 @@ func DeletePipelineRC(t testing.TB, pipeline string) {
 		return nil
 	})
 }
+
+// PachdDeployment finds the corresponding deployment for pachd in the
+// kubernetes namespace and returns it.
+func PachdDeployment(t testing.TB, namespace string) *apps.Deployment {
+	k := GetKubeClient(t)
+	result, err := k.AppsV1().Deployments(namespace).Get("pachd", metav1.GetOptions{})
+	require.NoError(t, err)
+	return result
+}
+
+func podRunningAndReady(e watch.Event) (bool, error) {
+	if e.Type == watch.Deleted {
+		return false, errors.New("received DELETE while watching pods")
+	}
+	pod, ok := e.Object.(*v1.Pod)
+	if !ok {
+		return false, errors.Errorf("unexpected object type in watch.Event")
+	}
+	return pod.Status.Phase == v1.PodRunning, nil
+}
+
+// WaitForPachdReady finds the pachd pods within the kubernetes namespace and
+// blocks until they are all ready.
+func WaitForPachdReady(t testing.TB, namespace string) {
```

Comment:
It'll be really nice to have a Go implementation of this logic. I think right now we have similar logic implemented in a few places, and maybe this will let us consolidate later? I'm thinking of:
- etc/reset.py
- etc/kube/start-minikube-vm.sh
- The hub backend, I think? (though I think the hub backend uses kubectl in some of the places that this uses the kubernetes Go client?)